polars/polars-sql/src/sql_expr.rs

```diff
@@ -432,6 +428,34 @@ impl SqlExprVisitor<'_> {
                 None => polars_bail!(ComputeError: "ELSE expression is required"),
             };
 
+            if let Some(operand_expr) = operand {
+                let first_operand_expr = self.visit_expr(operand_expr)?;
```

Comment:
This feels just a tad too easy to be optimal. Will CSE optimize multiple uses of `operand_expr`

Reply:
Yes! In the next release it will. :)